### PR TITLE
[CI] added new workflows to automatically label pull request according to file changed.

### LINKED ISSUE #4234: [CI] A workflow for automatically labeling a pull request according to files changed by the contributor.
#### Current Behavior
Even after this [pull request](https://github.com/meshery/meshery/pull/4151) due to issues regarding the `GITHUB_TOKEN`, the labels are not being assigned automatically. Changing the token could have solved the issue but doing so would have caused problems while releasing drafts. Hence a new workflow is required.

#### Desired Behavior
The pull request should be labeled automatically according to the files changed by the contributor. 
Example: If a contributor has changed contents from the docs folder and created a pull request, then the pull request should be labeled `area/docs` automatically.

#### Implementation
We can use the [Pull Request Labeler ](https://github.com/actions/labeler)action of GitHub to do so. We have to create two `.yaml` files one inside the `.github/workflow` folder and another inside the `.github` folder.

#### Acceptance Tests

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by asubedy**
**Description**
This pull request now will help in labeling any future pull request according to the files changed by the contributor. 
This PR fixes #4234 

**Notes for Reviewers**
I have deleted [previous workflow](https://github.com/meshery/meshery/pull/4151) on release drafter for auto-labeler. Did this so that there will be no unnecessary checks going on 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@asubedy we might want to walk through this on a call. I'm not sure that this change is desired. @DelusionalOptimist?

**Comment by asubedy** [comment]
@leecalcote sure we can set up a call for this. I had discussed this with @DelusionalOptimist on slack. As stated on issue #4234  The GitHub token that we are using now prevents giving labels automatically, and we can not change it as it will affect the release procedure. Thus a need for a new workflow.

**Comment by DelusionalOptimist** [comment]
Hmm... This dicussion can have a place in tomorrow's Meshery dev call's agenda :+1:

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+area/docs:
+- docs/**/*
+area/ci:
+- '.github/**/*'      
+component/ui:
+- 'ui/**/*'
+- 'provider-ui/**/*'
+component/mesheryctl:
+- 'mesheryctl/**/*'
```
Comment (ramrodo) [inline .github/labeler.yml:9]: Add blank line at the end of the file

File: .github/workflows/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+on:
+- pull_request_target
+
+jobs:
+  triage:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/labeler@v3
+      with:
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (ramrodo) [inline .github/workflows/labeler.yml:11]: Add a blank line at the end of the file

**Comment by ramrodo** [comment]
@asubedy did you all solve the problem related to the Github token?

**Comment by asubedy** [comment]
@ramrodo I did mention the issues to @leecalcote on the slack channel but I am still waiting for a reply. But for the existing workflow to work we have to change the token, but again it may cause issues regarding permission for the release drafter. So right now I am not sure how should I proceed.

File: .github/workflows/labeler.yml (hunk 2 above)
Comment (asubedy) [reply .github/workflows/labeler.yml:11]: Yep, will do that.

**Comment by ramrodo** [comment]
@asubedy I think we should not expect this workflow is triggered on forks according to https://github.com/actions/labeler/issues/20.  Also, if we are not sure if this may affect the `release-drafter` workflow, undo the changes related to this and keep only those related to the `labeler`.

**Pull request merged**